File: day1/task2/solver2.py
import sys

def solver(*args):
    freqs = {0: True} 
    acc = 0
    while True:
        for arg in args:
            acc += int(arg)
            if acc in freqs:
                return acc
            freqs[acc] = True 
    return acc

print(solver(*sys.argv[1:]))

File: day1/task2/solver2.js
function solver(args){
    let freqs = {'0':1}
    let acc = 0
    while(true){
        for(var i = 0, len = args.length; i < len; i++){
            acc += args[i]
            if(freqs[acc])
                return acc
            freqs[acc] = 1
        }
    }
}


console.log(solver(process.argv.slice(2, process.argv.length).map(val=>{
    return Number(val)
})))